osu.Game/Rulesets/UI/RulesetInputManager.cs

```diff
@@ -147,9 +149,9 @@ protected override bool Handle(UIEvent e)
 
         protected override bool HandleMouseTouchStateChange(TouchStateChangeEvent e)
         {
-            if (mouseDisabled.Value)
+            if (tapsDisabled.Value)
             {
-                // Only propagate positional data when mouse buttons are disabled.
+                // Only propagate positional data when taps are disabled.
                 e = new TouchStateChangeEvent(e.State, e.Input, e.Touch, false, e.LastPosition);
             }
```

Comment:
This isn't doing much. All the rulesets now support touch natively and don't need mouse mouse emulation. And therefore completely ignore this input path. Possibly still relevant for custom rulesets.

Reply:
This path is specific to the osu! ruleset, and it should be removed now that osu! handles touch input appropriately with the `OsuTouchInputMapper` component (as well as correctly ignoring taps while still maintaining position, without doing it weirdly like in the code highlighted here).

But that should be done in a separate PR to keep review complexity to a minimum.

Reply:
> This path is specific to the osu! ruleset

How so? I don't understand. This should apply to any ruleset which supports mouse but not explicitly touch?

Reply:
Hmm, I guess freeform custom rulesets would end up with a non-functioning setting that way...fair enough.